src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -1799,6 +1807,33 @@ bool core::check_tx_semantic(const transaction& tx, bool keeped_by_block) const
     return true;
 }
 //-----------------------------------------------------------------------------------------------
+void core::check_service_node_ip_address() {
+    if (!m_service_node || m_has_ip_check_disabled || service_node_list.debug_allow_local_ips) {
+        return;
+    }
+
+    auto service_node_ip = epee::string_tools::get_ip_string_from_int32(m_sn_public_ip);
+
+    m_omq->connect_remote(
+            oxenmq::address{
+                    "tcp://{}:{}"_format(service_node_ip, m_quorumnet_port),
+                    tools::view_guts(m_service_keys.pub_x25519)},
+            [&](auto conn) {
+                m_omq->request(
+                        conn,
+                        "ping.ping",
+                        [this, conn](bool success, const std::vector<std::string>& data) {
+                            m_omq->disconnect(conn, 0s);
+                            if (!success || data.empty())
+                                log::warning(
+                                        globallogcat, "Unable to ping configured service node IP!");
```

Comment:
It might be nice to include the IP/port here, so that you can see the configured address in the warning and can immediately say "oh, that's not right!" and fix it.  Something like:
```
globallogcat, "Unable to ping configured service node address ({}:{})!",
epee::string_tools::get_ip_string_from_int32(m_sn_public_ip),
m_quorumnet_port);
```